src/test/java/uk/gov/hmcts/reform/idam/ApplicationExecutorTest.java

```diff
@@ -44,4 +48,23 @@ void shouldNotRunAnyIfBothDisabled() {
         verify(idamUserDisposerService, times(0)).run();
         verify(securityUtil, times(0)).generateTokens();
     }
+
+    @Test()
+    void shouldCatchExceptionIfThrowExceptionTrue() {
+
+        // given
+        ReflectionTestUtils.setField(executor, "isDisposerEnabled", true);
+        Mockito.doThrow(new IllegalArgumentException("Exception to test alert"))
+            .when(idamUserDisposerService).run();
+
+        // when
+        executor.run(applicationArguments);
+
+        // then
+        verify(idamUserDisposerService, times(1)).run();
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class,
+                                                          () -> idamUserDisposerService.run());
+        assertThat(exception).hasMessageContaining("Exception to test alert");
```

Comment:
You are writing code here. You have mock, that you set to throw an exception on invocation, and then you invoke, and check that your mock threw an exception. There is no actual code testing here.